src/main/kotlin/calculator/Calculator.kt

```diff
@@ -0,0 +1,71 @@
+package calculator
+
+fun String?.tokenize(): List<String> = this?.trim()?.split("\\s+".toRegex())?.filter { it != "" }
+    ?: throw IllegalArgumentException("Null cannot be tokenized")
+
+fun List<String>.parse(): Expression {
+    fun parse(exp: Expression, tokenList: List<String>): Expression {
```

Comment:
```suggestion
    tailrec fun parse(exp: Expression, tokenList: List<String>): Expression {
```
코틀린은 [꼬리 재귀 최적화](https://kotlinlang.org/docs/functions.html#tail-recursive-functions)를 지원합니다 😮 

실제로 어떻게 컴파일 되는지는 [교육 자료](https://edu.nextstep.camp/s/MFvxefXA/ls/rclWyBPz)에서 확인하실 수 있습니다 😄

Reply:
재귀 함수를 실제 코드에 적용하기 시작한지는 얼마 되지 않아서..
코드를 짜면서도 꼬리 재귀인지 아닌지 판단할 새도 없이 일단 구현부터 하고 봤네요. :)
tailrec 꼭 기억하겠습니다.